MoltenVK/MoltenVK/GPUObjects/MVKQueryPool.mm

```diff
@@ -183,9 +185,16 @@
 		_availabilityLock.lock();
 		cmdEncoder->setComputeBytes(mtlComputeCmdEnc, _availability.data(), _availability.size() * sizeof(Status), 5);
 		_availabilityLock.unlock();
+
 		// Run one thread per query. Try to fill up a subgroup.
-		[mtlComputeCmdEnc dispatchThreadgroups: MTLSizeMake(max(queryCount / mtlCopyResultsState.threadExecutionWidth, NSUInteger(1)), 1, 1)
-						  threadsPerThreadgroup: MTLSizeMake(min(NSUInteger(queryCount), mtlCopyResultsState.threadExecutionWidth), 1, 1)];
+		NSUInteger threadCount = NSUInteger(queryCount);
+		NSUInteger threadExecutionWidth = mtlCopyResultsState.threadExecutionWidth;
+		NSUInteger tgWidth = min(threadCount, threadExecutionWidth);
+		NSUInteger tgCount = threadCount / threadExecutionWidth;
```

Comment:
What happens when there are fewer queries than will fill up an SIMD-group?

Reply:
Is your concern about passing the right values to `dispatchThreadgroups:threadsPerThreadgroup:`?

If so, I think this

    NSUInteger threadCount = NSUInteger(queryCount);
    NSUInteger tgWidth = min(threadCount, threadExecutionWidth);

combined with  the guard code in the shader

    if (query >= numQueries) { return; }

 should cover that?

Or are you concerned that there is something deeper in the hardware that will run into trouble?

Reply:
> Is your concern about passing the right values to `dispatchThreadgroups:threadsPerThreadgroup:`?

Yes.

What I'm trying to say is, if `queryCount` is less than `threadExecutionWidth`, then the division will come out zero, which isn't what we want. This division probably needs to round up.

Reply:
> This division probably needs to round up.

Ah. Yes. The line following it does that.

Reply:
How did I not notice that...?

Ah well. Never mind, then.